### ISSUE TITLE
Internal error on import psycopg module

### ISSUE THREAD

**Issue by louisbertrand** [PyCQA/pylint#10415]
Crash reported by the Thonny Assistant even though the Python script worked correctly.
```
Warnings
May be ignored if you are happy with your program.
testing_psycopg.py
Line 1 : INTERNAL ERROR when analyzing the code: /home/louis/devel/safecast stuff/regional_map/scripts/testing_psycopg.py: Fatal error while checking '/home/louis/devel/safecast stuff/regional_map/scripts/testing_psycopg.py'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in '/home/louis/.var/app/org.thonny.Thonny/cache/pylint/pylint-crash-2025-06-05-04-52-12.txt'.
Program analyzer internal error. Used when an unexpected error occurred while building the Astroid representation. This is usually accompanied by a traceback. Please report such errors !
```
Thonny 4.1.7, Linux, Python 3.12.10, Tk 8.6.15 (Distro: Linux Mint 22.1 Xia base: Ubuntu 24.04 noble)

The **psycopg** module was installed:
```
Installed version: 3.2.9
Installed to: /var/data/python/lib/python3.12/site-packages

Latest stable version: 3.2.9
Summary: PostgreSQL database adapter for Python
Author: Daniele Varrazzo
License: GNU Lesser General Public License v3 (LGPLv3)
Homepage: https://psycopg.org/psycopg3/
PyPI page: https://pypi.org/project/psycopg/
Requires: backports.zoneinfo>=0.2.0; python_version < "3.9", typing-extensions>=4.6; python_version < "3.13", tzdata; sys_platform == "win32"
```
Secondary issue: The crash report file was not created until I manually created the directory
  .var/app/org.thonny.Thonny/cache/pylint
But that's probably a Thonny issue.

[pylint-crash-2025-06-05-05-25-37.txt](https://github.com/user-attachments/files/20607633/pylint-crash-2025-06-05-05-25-37.txt)

**Comment by jacobtylerwalls** [comment]
Use pylint 3+ if running python 3.12.

(triagers: see `visit_typealias` in the traceback)
